gk/fib.c

```diff
@@ -1793,6 +1797,8 @@ fillup_gk_fib_dump_entry(struct gk_fib_dump_entry *dentry, struct gk_fib *fib)
 
 #define CTYPE_STRUCT_FIB_DUMP_ENTRY_PTR "struct gk_fib_dump_entry *"
 
+#define MEMBLOCK_SIZE	(8 * 1024)
```

Comment:
It is not a good thing that the amount of memory to pass to a single call of the Lua callback varies that much. It is like passing the whole FIB table with a single call, that is, the size would vary with the size of the table; this no iterator comparison is here just to make my point.

The review of this pull request is already long. Keep this pull request, but close it. This will be helpful for @cjdoucette to have it as an example of how to use memblocks. Open a new pull request for the same goal and using the code here as a start, but the following changes are needed:

1. `list_ipv4_fib_entries()` and `list_ipv6_fib_entries()` should skip FIB entries of the type `GK_FWD_NEIGHBOR_FRONT_NET` and `GK_FWD_NEIGHBOR_BACK_NET`.
2. Thanks to the change before, we no longer need `struct gk_fib_ether_dump_entry` since its fields can become part of `struct gk_fib_dump_entry`.
3. The Lua names `list_fib4_entries()` and `list_fib6_entries()` should be renamed to `list_gk_fib4()` and `list_gk_fib6()` to emphasize that we are dealing with the GK block only. This is preparing the code for future patches, for example, Cody is going to add `list_lls_arp()` and `list_lls_nd()` soon and, eventually, the GT block may have a FIB as well.
4. Add `list_gk_neighbors4()` and `list_gk_neighbors6()` to list the entries of the two fib entries `GK_FWD_NEIGHBOR_FRONT_NET` and `GK_FWD_NEIGHBOR_BACK_NET`. This will bound the amount of memory needed to call this new Lua callback. Thanks to this step, we won't need memblocks for now.

Reply:
To simplify the review of the new pull request, you can add step 4 as a second patch since the review of the first patch, which will be a variation of the current patch here, should be easy.

Reply:
Will do.